docs/adr/adr-005-data-retention.md

```diff
@@ -0,0 +1,201 @@
+data could result in capacity issues due to the rapid increase in data being
+produced. E.g. 10 months ago we only produced ~54,000 `DigestRunSubscriber`
+records each day, which is roughly ~12% of the amount generated each day
+currently.
+
+Another aspect of storing lots of unused data is that it complicates our
+auditing and analysis. An example of how dead data can skew our analytics can
+be found when we look at subscriber lists. We have over ~73,000 of these lists
+with active subscribers, but in the last week alone, we've accumulated 1139
+lists which don't have any subscriptions at all (ended or active). These dead
+lists prevent us from doing simple things easily, like determining which lists
+are actually in use.
+
+[archived]: ...
+[deleted]: ...
+[nullified]: ...
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-initial-data-retention.md
+[migrate subscribers to daily digest emails]: https://github.com/alphagov/email-alert-api/pull/1352
+
+---
+
+## Decision
+
+The decision we came to was to implement a [worker] which runs periodically
+every day at [midday] to remove data we classify as historic. Removing this
+data was deemed as non urgent and so deleting once a day, in hours, seemed
+sufficient.
+
+To deem data as historic we had to be ensure that the system no longer
+uses the data, and we don't need the data for future analytics.
+
+Below are the decisions we made.
+
+### Max retention of 1 year for everything that's unused by the system
+
+There is no evidence of needing unused data for analytics beyond 1 year and so
+we can classify this old data as historic and remove it.
+
+We can split this unused data into two categories.
+
+Event data - which is only used for a short period after being created, and
+subsequently for analytics, which includes:
+
+* ContentChange & Message - removed once their `created_at` value is over a
+  year old.
+
+* MatchedContentChange & MatchedMessage - cascaded when their parent content
+  change / message is deleted.
+
+* DigestRun - removed once their `created_at` value is over a year old.
+
+* DigestRunSubscriber - cascaded when their parent digest run gets deleted.
+
+Subscriber data - which is used indefinitely until it's marked as "ended", which
+includes:
+
+* Subscription - removed when they've been ended for over a year, based on
+  their `ended_at` value. Once a subscription is ended it [can't currently be
+  made active again] and so can be deleted.
+
+* SubscriberList - removed when they have no active subscriptions and their
+  latest, ended subscription is over a year old.
+
+* Subscriber - removed once their `created_at` is over a year old and they
+  don't have any subscriptions (active or ended).
+
+### Shorter retention of 7 days for subscriber lists with no subscriptions
+
+We also remove subscriber lists which have no subscriptions (active or ended)
+and are over 7 days old. As mentioned in the context section, dead subscriber
```

Comment:
```suggestion
and are over 7 days old. As mentioned in the context section, unused subscriber
```

We use the term "dead" elsewhere to refer to lists that won't ever have matched changes against them.